test/e2e/framework/pods.go

```diff
@@ -29,70 +29,74 @@ import (
 	. "github.com/onsi/gomega"
 )
 
-// TODO: Consolidate pod-specific framework functions here.
-
 // Convenience method for getting a pod client interface in the framework's namespace.
-func (f *Framework) PodClient() unversioned.PodInterface {
-	return f.Client.Pods(f.Namespace.Name)
+func (f *Framework) PodClient() *PodClient {
+	return &PodClient{
+		f:            f,
+		PodInterface: f.Client.Pods(f.Namespace.Name),
+	}
 }
 
-// Create a new pod according to the framework specifications, and wait for it to start.
-// Returns the server's representation of the pod.
-func (f *Framework) CreatePod(pod *api.Pod) *api.Pod {
-	p := f.CreatePodAsync(pod)
-	ExpectNoError(f.WaitForPodRunning(p.Name))
-	return p
+type PodClient struct {
```

Comment:
Just want to make the pod client more straightforward to use (similar with the original pod client).
And make it easier to refactor current test.